quran/api/tests/views/test_surah_list_retrieve_view.py

```diff
@@ -0,0 +1,35 @@
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APITestCase
+
+from quran.api.serializers import SurahListSerializer, SurahDetailsSerializer
+from quran.models import Surah
+
+
+class TestSurahListRetrieveView(APITestCase):
+
+    def setUp(self):
+        self.list_url = reverse("api:quran-list")
+        self.surah_number = 1
+        self.details_url = reverse("api:quran-detail", kwargs={"number": self.surah_number})
+
+    def test_list_url(self):
+        self.assertEqual(self.list_url, "/api/quran/surah/")
```

Comment:
What's the values of the test?
You are using reverse urls in order to avoid hard-coding them. And here you are hard-coding them. So once the url changes, your tests are broken.
On the other hand, I can see the logic of protecting yourself from changing the API by mistake.

It's quite controversial.